File: codesamples/sample.java
public static class RemindUserWorkflowImpl implements RemindUserWorkflow {
  public void EmailUser(String userId, int[] intervals) {
    // Send reminder emails, e.g. after 1, 7, and 30 days
    for (int interval : intervals) {   
      Workflow.sleep(Duration.ofDays(interval)); // Sleep for days!
      activities.sendEmail(interval, userId);    // Activities retried by default!
    }
    // Easily cancelled when user unsubscribes
  }
}

File: codesamples/sample.php
class RemindUserWorkflow implements RemindUserWorkflowInterface {
  public function emailUser($userID, $intervals) {
      // Send reminder emails, e.g. after 1, 7, and 30 days
      foreach ($intervals as &$interval) {
          yield Workflow::timer($interval * DAYS);                  // Sleep for days!
          yield $this->userActivity->sendEmail($interval, $userID); // Activities retried by default!
      }
      // Easily modifiable to cancel if the user unsubscribes
  }
}